lib/puppet/type/cloudwatch_alarm.rb

```diff
@@ -0,0 +1,91 @@
+Puppet::Type.newtype(:cloudwatch_alarm) do
+  @doc = 'type representing an AWS CloudWatch Alarm'
+
+  ensurable
+
+  newparam(:name, namevar: true) do
+    desc 'the name of the alarm'
```

Comment:
Sort of a nitpick, but I'd like to see the first letter capitalized and the phrase ended with a period. The `ensure` property has pregenerated documentation that uses that style, so it's more pleasant to see consistent style when running `puppet describe`:

```bash
$ puppet describe cloudwatch_alarm --modulepath ..

cloudwatch_alarm
================
type representing an AWS CloudWatch Alarm


Parameters
----------

- **alarm_actions**
    the actions to trigger when the alarm triggers

- **comparison_operator**
    the operator to use to test the metric

- **dimensions**
    the dimensions to filter the alerm by

- **ensure**
    The basic property that the resource should be in.
    Valid values are `present`, `absent`.

- **evaluation_periods**
    the number of checks to use to confirm the alarm
...
```